src/clib/cool1d_cloudy_old_tables_g.cpp

```diff
@@ -0,0 +1,352 @@
+//===----------------------------------------------------------------------===//
+//
+// See the LICENSE file for license and copyright information
+// SPDX-License-Identifier: NCSA AND BSD-3-Clause
+//
+//===----------------------------------------------------------------------===//
+///
+/// @file
+/// Implements the cool1d_cloudy_old_tables_g function
+///
+//===----------------------------------------------------------------------===//
+
+// This file was initially generated automatically during conversion of the
+// cool1d_cloudy_old_tables_g function from FORTRAN to C++
+
+#include <cstdio>
+#include <cmath>
+#include <vector>
+
+#include "grackle.h"
+#include "fortran_func_decls.h"
+#include "fortran_func_wrappers.hpp"
+#include "utils-cpp.hpp"
+
+#include "cool1d_cloudy_old_tables_g.hpp"
+
+namespace grackle::impl {
+
+void cool1d_cloudy_old_tables_g(double* rhoH, double* metallicity,
+                                double* logtem, double* edot, double comp2,
+                                double dom, double zr, gr_mask_type* itmask,
+                                chemistry_data* my_chemistry,
+                                chemistry_data_storage* my_rates,
+                                grackle_field_data* my_fields,
+                                IndexRange idx_range) {
```

Comment:
would you mind making a few simplifications to the signature?

Namely:

- could you replace `chemistry_data_storage* my_rates` with something like `cloudy_data cloudy_table`,
- I think it may make more sense to replace `grackle_field_data* my_fields` with `const gr_float* density` and `const gr_float* eint` so that the code becomes more easy to unit test
   - there's a few spots where use `grackle_fields->grid_dimension[0]`. We can replace those with `idx_range->i_stop`